entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/SiteEntityProvider.java

```diff
@@ -309,7 +310,11 @@ public ActionReturn handleGroups(EntityView view, Map<String, Object> params) th
         // check if the user can access site
         isAllowedAccessSite(site);
 
-        List<EntityGroup> groups = site.getGroups().stream().map(EntityGroup::new).collect(Collectors.toList());
+        List<EntityGroup> groups = securityService.isSuperUser() ?
+                site.getGroups().stream().map(EntityGroup::new).collect(Collectors.toList())
+                :
+                site.getGroupsWithMember(developerHelperService.getCurrentUserId()).stream()
+                .map(EntityGroup::new).sorted((group, other) -> StringUtils.compare(group.getTitle(), other.getTitle())).collect(Collectors.toList());
```

Comment:
i believe we have a special Comparator to handle group name sorting.

because default alpha sorting will do this:

Group 1
Group 10
Group 11
Group 2
Group 3

Maybe it's this one? 

./kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/AlphaNumericComparator.java

Reply:
SAK-41426

Reply:
Done!